src/__swaps__/utils/swaps.ts

```diff
@@ -233,139 +232,53 @@ export function trimTrailingZeros(value: string) {
-  roundingMode?: 'up' | 'down' | 'none';
-  precisionAdjustment?: number;
-  isStablecoin?: boolean;
-  stripSeparators?: boolean;
-}): string {
-  'worklet';
-
-  function calculateDecimalPlaces(): {
-    minimumDecimalPlaces: number;
-    maximumDecimalPlaces: number;
-  } {
-    if (nativePrice === 0) {
-      return {
-        minimumDecimalPlaces: 0,
-        maximumDecimalPlaces: MAXIMUM_SIGNIFICANT_DECIMALS,
-      };
-    }
-
-    const unitsForOneCent = 0.01 / nativePrice;
-    if (unitsForOneCent >= 1) {
-      return {
-        minimumDecimalPlaces: 0,
-        maximumDecimalPlaces: 0,
-      };
-    }
-
-    return {
-      minimumDecimalPlaces: isStablecoin ? STABLECOIN_MINIMUM_SIGNIFICANT_DECIMALS : 0,
-      maximumDecimalPlaces: Math.max(
-        Math.ceil(Math.log10(1 / unitsForOneCent)) + (precisionAdjustment ?? 0),
-        isStablecoin ? STABLECOIN_MINIMUM_SIGNIFICANT_DECIMALS : 0
-      ),
-    };
-  }
-
-  const { minimumDecimalPlaces, maximumDecimalPlaces } = calculateDecimalPlaces();
-
-  let roundedAmount;
-  const factor = Math.pow(10, maximumDecimalPlaces) || 1; // Prevent division by 0
-
-  // Apply rounding based on the specified rounding mode
-  if (roundingMode === 'up') {
-    roundedAmount = divWorklet(ceilWorklet(mulWorklet(amount, factor)), factor);
-  } else if (roundingMode === 'down') {
-    roundedAmount = divWorklet(floorWorklet(mulWorklet(amount, factor)), factor);
-  } else if (roundingMode === 'none') {
-    roundedAmount = toFixedWorklet(amount, maximumDecimalPlaces);
-  } else {
-    // Default to normal rounding if no rounding mode is specified
-    roundedAmount = divWorklet(roundWorklet(mulWorklet(amount, factor)), factor);
-  }
-
-  // Format the number to add separators and trim trailing zeros
-  const numberFormatter = new Intl.NumberFormat('en-US', {
-    minimumFractionDigits: minimumDecimalPlaces,
-    maximumFractionDigits: maximumDecimalPlaces,
-    useGrouping: !stripSeparators,
-  });
-
-  return numberFormatter.format(Number(roundedAmount));
-}
-
 export function niceIncrementFormatter({
-  incrementDecimalPlaces,
   inputAssetBalance,
   inputAssetNativePrice,
-  niceIncrement,
   percentageToSwap,
   sliderXPosition,
   stripSeparators,
   isStablecoin = false,
 }: {
-  incrementDecimalPlaces: number;
   inputAssetBalance: number | string;
   inputAssetNativePrice: number;
-  niceIncrement: number | string;
   percentageToSwap: number;
   sliderXPosition: number;
   stripSeparators?: boolean;
   isStablecoin?: boolean;
 }) {
   'worklet';
+  const niceIncrement = findNiceIncrement(inputAssetBalance);
+  const incrementDecimalPlaces = countDecimalPlaces(niceIncrement);
 
   if (percentageToSwap === 0 || equalWorklet(niceIncrement, 0)) return '0';
   if (percentageToSwap === 0.25) {
     const amount = mulWorklet(inputAssetBalance, 0.25);
     return valueBasedDecimalFormatter({
       nativePrice: inputAssetNativePrice,
+      niceIncrementMinimumDecimals: incrementDecimalPlaces,
```

Comment:
This is point #4: Added an optional param to valueBasedDecimalFormatter that the niceIncrementFormatter can pass in such that it guarantees that the valueBasedDecimalFormatter is always at least +1 more than the niceIncrementFormatter number of decimals. This helps resolve scenarios where the "jumps" between increments might end up rounding funnily on the snapping points.